src/ggl-setup.py

```diff
@@ -0,0 +1,524 @@
+        if not _check_group_exists(group):
+            subprocess.run(['sudo', 'groupadd', group], check=True)
+            print(f"Successfully created {group} group")
+
+        # Create the user and add to the group
+        if not _check_user_exists(user):
+            subprocess.run(['sudo', 'useradd', '-Ng', group, user], check=True)
+            print(f"Successfully created {user} user")
+        return True
+
+    except subprocess.CalledProcessError as e:
+        print(f"Command failed: {e.cmd}. Return code: {e.returncode}")
+        return False
+    except Exception as e:
+        print(f"Error when adding user and group: {str(e)}")
+        return False
+
+
+def _delete_user_and_group(user: str, group: str):
+    try:
+        # Delete the user
+        if _check_user_exists(user):
+            subprocess.run(['sudo', 'userdel', user], check=True)
+            print(f"Successfully deleted {user} user")
+
+        # Delete the group
+        if _check_group_exists(group):
+            subprocess.run(['sudo', 'groupdel', group], check=True)
+            print(f"Successfully deleted {group} group")
+        return True
+
+    except subprocess.CalledProcessError as e:
+        print(f"Command failed: {e.cmd}. Return code: {e.returncode}")
+        return False
+    except Exception as e:
+        print(f"Error when deleting user and group: {str(e)}")
+        return False
+
+
+def _check_group_exists(group: str) -> bool:
+    try:
+        grp.getgrnam(group)
+        return True
+    except KeyError:
+        return False
+
+
+def _check_user_exists(user: str) -> bool:
+    try:
+        pwd.getpwnam(user)
+        return True
+    except KeyError:
+        return False
+
+
+def _build_with_cmake() -> bool:
+    original_dir = os.getcwd()
+    try:
+        # CMake configure command
+        cmake_cmd = [
+            'cmake', '-B', 'build', '-D', 'CMAKE_INSTALL_PREFIX=/usr/local',
+            '-D', 'CMAKE_BUILD_TYPE=MinSizeRel', '-D', 'CMAKE_BUILD_TYPE=Debug',
+            '-DGGL_LOG_LEVEL=DEBUG'
+        ]
+
+        subprocess.run(cmake_cmd, check=True)
+
+        # Change the directory to build
+        os.chdir("build")
+        build_cmd = ['make', '-C', 'build', '-j$(nproc)']
+        subprocess.run(build_cmd, shell=True, check=True)
+        os.chdir(original_dir)
+
+        # Install command
+        install_cmd = ['sudo', 'make', '-C', 'build', 'install']
+        subprocess.run(install_cmd, check=True)
+
+        print("Successfully completed the build process")
+        return True
+
+    except subprocess.CalledProcessError as e:
+        print(f"Error during build process: {e}")
+        return False
+    except Exception as e:
+        print(f"Unexpected error: {e}")
+        return False
+    finally:
+        os.chdir(original_dir)
+
+
+def _tes_setup(device_cert: str, private_key: str) -> bool:
+
+    # Save the key and certificates
+    try:
+        device_result = _create_file(DEVICE_PATH, device_cert)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

 The absolute path starts at the root and moves down through each file. While traversing the path, if the folder name or location is different than the file path mentioned in the code, it causes a break. Since most of the computers have different directory structures, using absolute path in your code is likely to cause issues in your computational environment. We recommend using relative paths, because it only relies on relevant part of the directory structure being the same making it reliable to work with in computational environment

Similar issue at line numbers 327 and 334.